retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/RetrofitClassAwareMeterIdPrefixFunction.java

```diff
@@ -0,0 +1,190 @@
+        buildTags(tagListBuilder, log);
+        RequestMetricSupport.appendHttpStatusTag(tagListBuilder, log);
+        return new MeterIdPrefix(name, tagListBuilder.build());
+    }
+
+    private void buildTags(ImmutableList.Builder<Tag> tagsBuilder, RequestLog log) {
+        final Invocation invocation = InvocationUtil.getInvocation(log);
+
+        tagsBuilder.add(Tag.of(serviceTagName, serviceName));
+        if (invocation != null) {
+            final String methodName = invocation.method().getName();
+            final List<Tag> methodTags = methodNameToTags.get(methodName);
+            if (methodTags != null) {
+                tagsBuilder.addAll(methodTags);
+                return;
+            }
+        }
+
+        tagsBuilder.add(Tag.of(HTTP_METHOD_TAG_NAME, log.method().name()));
+        tagsBuilder.addAll(DEFAULT_METHOD_TAGS);
+    }
+
+    @VisibleForTesting
+    static Map<String, List<Tag>> defineTagsForMethods(Class<?> serviceClass) {
+        final ImmutableMap.Builder<String, List<Tag>> methodNameToTags = ImmutableMap.builder();
+
+        final Method[] declaredMethods = serviceClass.getDeclaredMethods();
+        for (final Method clientServiceMethod : declaredMethods) {
+            for (final Annotation annotation : clientServiceMethod.getDeclaredAnnotations()) {
+                if (!RETROFIT_ANNOTATIONS.contains(annotation.annotationType())) {
+                    continue;
+                }
+
+                final ImmutableList.Builder<Tag> tags = ImmutableList.builder();
+                tags.addAll(tagsFromAnnotation(annotation));
+                tags.add(Tag.of(METHOD_TAG_NAME, clientServiceMethod.getName()));
+                methodNameToTags.put(clientServiceMethod.getName(), tags.build());
+            }
+        }
+
+        return methodNameToTags.build();
+    }
+
+    private static List<Tag> tagsFromAnnotation(Annotation annotation) {
+        final String httpMethod;
+        final String httpPath;
+        if (annotation.annotationType().equals(HTTP.class)) {
+            final HTTP http = (HTTP) annotation;
+            httpMethod = http.method();
+            httpPath = http.path();
+        } else {
+            final Method valueMethod;
+            try {
+                valueMethod = annotation.getClass().getMethod("value");
+                httpPath = (String) valueMethod.invoke(annotation);
+            } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                // Should never happen on valid Retrofit client.
+                throw new IllegalStateException("Unexpected retrofit annotation: " +
+                                                annotation.annotationType(), e);
+            }
+            httpMethod = annotation.annotationType().getSimpleName();
+        }
+
+        return ImmutableList.of(
+                Tag.of(HTTP_METHOD_TAG_NAME, httpMethod),
+                Tag.of(PATH_TAG_NAME, httpPath)
```

Comment:
How about 
> This decorator currently use `path` and `httpMethod` labels whether Armeria server metrics provide `route` label that contains path and http method at the same time

Makes sense to unify? Or just add another label :thinking:  I feel that it maybe easier to make queries when path and httpMethod separated

Reply:
Agreed. Separate tags sounds good to me.